File: haskellStage9/src/TicketRegistry.hs
module TicketRegistry (
    TicketRegistry,
    emptyRegistry,
    addTicket,
    findTicket,
    sizeRegistry
) where

import Ticket
import QTree (QTree, emptyQTree, insertQTree, searchQTree, sizeQTree)

type TicketRegistry = QTree Ticket

emptyRegistry :: TicketRegistry
emptyRegistry = emptyQTree

addTicket :: Ticket -> TicketRegistry -> TicketRegistry
addTicket = insertQTree

findTicket :: Ticket -> TicketRegistry -> Bool
findTicket = searchQTree

sizeRegistry :: TicketRegistry -> Int
sizeRegistry = sizeQTree

File: Stage10/TicketRegistry.cpp
#include "TicketRegistry.h"

bool TicketRegistry::addTicket(const Ticket& ticket) {
    return storage.insert(ticket);
}

bool TicketRegistry::findTicket(int ticketId) const {
    return storage.search(ticketId);
}

size_t TicketRegistry::size() const {
    return storage.size();
}